src/NServiceBus.Gateway/Deduplication/LegacyDeduplicationStorageConfiguration.cs

```diff
@@ -22,12 +24,19 @@ public override void Setup(ReadOnlySettings settings)
                 throw new Exception("No persistence configured, please configure one that supports gateway deduplication storage or provide a deduplication storage via the 'endpointConfiguration.Gateway' API.");
             }
 
+            if (settings.IsFeatureEnabled(typeof(InMemoryGatewayPersistence)))
+            {
+                log.WarnFormat("Endpoint is configured to use the legacy in-memory gateway deduplication storage. This storage is going to be obsoleted in future versions of NServiceBus. Use the gateway's built-in in-memory storage instead: endpointConfiguration.UseGateway(new {0}());'", nameof(InMemoryDeduplicationConfiguration));
```

Comment:
```suggestion
                log.Warn($"Endpoint is configured to use the legacy in-memory gateway deduplication storage. This storage is going to be obsoleted in future versions of NServiceBus. Use the gateway's built-in in-memory storage instead: endpointConfiguration.UseGateway(new {nameof(InMemoryDeduplicationConfiguration)}());'");
```

This does not require a formatted structured error to be written as the type is static.